Sources/PluginLibrary/Array+Extensions.swift

```diff
@@ -0,0 +1,22 @@
+// Sources/PluginLibrary/Array+Extensions.swift - Additions to Arrays
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Array {
+  /// Like map, but calls the transform with the index and value.
+  func enumeratedMap<T>(_ transform: (Int, Element) throws -> T) rethrows -> [T] {
```

Comment:
☹️ 

I wonder why that's so much slower... `enumerated()` returns a lazily-evaluated sequence so it's not like it's creating a whole separate intermediate list.

Yeah, leave it and comment it then.